src/ILLink.CodeFix/DAMCodeFixProvider.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using ILLink.CodeFixProvider;
+using ILLink.RoslynAnalyzer;
+using ILLink.Shared;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace ILLink.CodeFix
+{
+	public class DAMCodeFixProvider : Microsoft.CodeAnalysis.CodeFixes.CodeFixProvider
+	{
+		public static ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => DynamicallyAccessedMembersAnalyzer.GetSupportedDiagnostics ();
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds => SupportedDiagnostics.Select (dd => dd.Id).ToImmutableArray ();
+
+		private protected static LocalizableString CodeFixTitle => new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersCodeFixTitle), Resources.ResourceManager, typeof (Resources));
```

Comment:
This line adds a new resource but I don't see the resource file being changed. So I'm guessing it will be missing

Reply:
I'm honestly not sure what this line does - what is the resource file?

Reply:
You have had some experience with them, we use resource files for the diagnostic messages (see the [SharedStrings.resx](https://github.com/dotnet/linker/blob/main/src/ILLink.Shared/SharedStrings.resx) file). We use them as a container for strings, that then we can use Localization to translate to different languages. In this case, in order to retrieve the string you need to add an entry to [Resources.resx](https://github.com/dotnet/linker/blob/main/src/ILLink.CodeFix/Resources.resx), which is the place where we place the other CodeFixer tittles.

Reply:
By the way is going to be interesting going forward, because unlike RequiresAttribute the DynamicallyAccessedMembersAttribute CodeFixer will place attributes in things that are not "the parent" so the message needs to be different accordingly. I would expect a matrix for how to generate the fix using the code fixer and decide the resource message to use in that same matrix, similar to what happens for diagnostic generation. For now, I guess is fine to have only one message.

Reply:
Got it, thanks! Yes, the DynamicallyAccessedMembers gets added to all sorts of locations which makes this code fixer a little more complicated to implement.